javapoet/src/test/java/com/palantir/javapoet/JavaFileTest.java

```diff
@@ -387,6 +388,169 @@ class Taco {
+
+                        record Taco<T>(List<T> names) {
+                        }
+                        """);
+    }
+
+    @Test
+    public void recordOneFieldImplementsInterface() {
+        String source = JavaFile.builder(
+                        "com.palantir.tacos",
+                        TypeSpec.recordBuilder("Taco")
+                                .recordConstructor(MethodSpec.constructorBuilder()
+                                        .addParameter(ParameterSpec.builder(String.class, "name")
+                                                .build())
+                                        .build())
+                                .addSuperinterface(Serializable.class)
+                                .build())
+                .skipJavaLangImports(true)
+                .build()
+                .toString();
+        assertThat(source)
+                .isEqualTo(
+                        """
+                        package com.palantir.tacos;
+
+                        import java.io.Serializable;
+
+                        record Taco(String name) implements Serializable {
+                        }
+                        """);
+    }
+
+    @Test
+    public void recordOneFieldWithAnnotation() {
+        String source = JavaFile.builder(
+                        "com.palantir.tacos",
+                        TypeSpec.recordBuilder("Taco")
+                                .recordConstructor(MethodSpec.constructorBuilder()
+                                        .addParameter(ParameterSpec.builder(String.class, "name")
+                                                .build())
+                                        .build())
+                                .addAnnotation(Deprecated.class)
+                                .build())
+                .skipJavaLangImports(true)
+                .build()
+                .toString();
+        assertThat(source)
+                .isEqualTo(
+                        """
+                        package com.palantir.tacos;
+
+                        @Deprecated
+                        record Taco(String name) {
+                        }
+                        """);
+    }
+
+    @Test
+    public void secondaryConstructorRecord() {
+        String source = JavaFile.builder(
+                        "com.palantir.tacos",
+                        TypeSpec.recordBuilder("Taco")
+                                .recordConstructor(MethodSpec.constructorBuilder()
+                                        .addParameter(ParameterSpec.builder(ClassName.get(String.class), "name")
+                                                .build())
+                                        .build())
+                                .addMethod(MethodSpec.constructorBuilder()
+                                        .addParameter(TypeName.INT, "number")
+                                        .addCode("this.name = $T.toString(number);", ClassName.get(Integer.class))
```

Comment:
This code produced here is invalid

```suggestion
                                        .addCode("this($T.toString(number));", ClassName.get(Integer.class))
```